### PR TITLE
[TIMOB-23206] iOS: Expose calendar source-details

### PR DESCRIPTION
**Pull request by AngelkPetkov**
[JIRA](https://jira.appcelerator.org/browse/TIMOB-23206)

### CODE REVIEW

File: iphone/Classes/TiCalendarCalendar.m [hunk 1]
```diff
@@ -291,6 +291,37 @@ -(NSString*)name
...
 }
 
+-(NSString*)sourceTitle
+{
+    if (![NSThread isMainThread]) {
+        __block id result;
+        TiThreadPerformOnMainThread(^{result = [[[self calendar] source] title];}, YES);
+        return [result autorelease];
+    }
+    return [[[self calendar] source] title];
```
Comment (hansemannn) [inline iphone/Classes/TiCalendarCalendar.m]: Why not just:
```objc
 __block id result;

TiThreadPerformOnMainThread(^{
    result = [[[self calendar] source] title];
}, YES);

return result;
```
If we are already on main-thread, `TiThreadPerformOnMainThread` will take of it and dispatch the block on it. Same for all other methods here.

**Comment by hansemannn** [comment]
Approved!

File: apidoc/Titanium/Calendar/Calendar.yml [hunk 2]
```diff
@@ -607,6 +607,48 @@ properties:
...
+
+  - name: SOURCE_TYPE_EXCHANGE
+    summary: A microsoft exchange calendar source.
+    type: Number
+    permission: read-only
+    platforms: [iphone, ipad]
+    since: "6.1.0"
+
+  - name: SOURCE_TYPE_CALDAV
+    summary: A calDev calendar source.
```
Comment (mrkiley) [inline apidoc/Titanium/Calendar/Calendar.yml]: summary typo here

**Comment by mrkiley** [comment]
Thanks for pushing this through guys!!! Awesome work!

**Pull request merged**